lib/topological_inventory/ansible_tower/operations/source.rb

```diff
@@ -11,9 +11,19 @@ class Source < TopologicalInventory::Providers::Common::Operations::Source
         private
 
         def connection_check
-          connection = ::TopologicalInventory::AnsibleTower::Cloud::Connection.new
-          connection = connection.connect(full_hostname(endpoint), authentication.username, authentication.password)
-          connection.api.version
+          if endpoint.receptor_node.present?
+            connection = ::TopologicalInventory::AnsibleTower::Receptor::Connection.new(
+              receptor_client,
+              endpoint.receptor_node,
+              account_number
+            )
+
+            connection.api.config
+          else
+            connection = ::TopologicalInventory::AnsibleTower::Cloud::Connection.new
+            connection = connection.connect(full_hostname(endpoint), authentication.username, authentication.password)
+            connection.api.version
+          end
```

Comment:
`ConnectionManager.connect` is doing this, with lazy initialization of `receptor_client` (except the api call)

Why did you choose `api.config` instead of `api.version` for on-premise?

Reply:
Cool! I updated it to use the `ConnectionManager.connect`, and I did api.version for both. I just did config out of habit. They both use the same API call now.